contrib/net/http/trace.go

```diff
@@ -63,24 +77,23 @@ func TraceAndServe(h http.Handler, w http.ResponseWriter, r *http.Request, cfg *
 	if spanctx, err := tracer.Extract(tracer.HTTPHeadersCarrier(r.Header)); err == nil {
 		opts = append(opts, tracer.ChildOf(spanctx))
 	}
-	span, ctx := tracer.StartSpanFromContext(r.Context(), "http.request", opts...)
-	rw, ddrw := wrapResponseWriter(w)
-	defer func() {
-		if ddrw.status == 0 {
-			span.SetTag(ext.HTTPCode, "200")
-		} else {
-			span.SetTag(ext.HTTPCode, strconv.Itoa(ddrw.status))
-		}
-		if ddrw.status >= 500 && ddrw.status < 600 {
-			span.SetTag(ext.Error, fmt.Errorf("%d: %s", ddrw.status, http.StatusText(ddrw.status)))
-		}
-		span.Finish(cfg.FinishOpts...)
-	}()
+	return tracer.StartSpanFromContext(r.Context(), "http.request", opts...)
+}
 
-	if appsec.Enabled() {
-		h = httpsec.WrapHandler(h, span, cfg.RouteParams)
+// FinishRequestSpan finishes the given HTTP request span with the standard list of HTTP request span tags.
+// Any further span finish option can be added with opts.
+func FinishRequestSpan(s tracer.Span, status int, opts ...tracer.FinishOption) {
+	var statusStr string
+	if status == 0 {
+		statusStr = "200"
+	} else {
+		statusStr = strconv.Itoa(status)
 	}
-	h.ServeHTTP(rw, r.WithContext(ctx))
```

Comment:
Where did these lines go? Should they be in `FinishRequestSpan`?

Reply:
This line moved and I think you got confused with the unified git diff view.
But now that Start/FinishRequestSpan were moved into their own internal package, the diff is now much clearer to read.

I'm resolving this comment and let you walk through the new diff of net/http/trace.go and let me know if anything 🙏